File: Problems/Mathematics/Day-02/sol/Adarsh Yadav/solution1.cpp
//https://codeforces.com/contest/1771/submission/355283451
//TC : O(n)

#include <bits/stdc++.h>
using namespace std;

int main() {
	int t;
	cin>>t;
	while(t--){
	    long long n;
	    cin>>n;
	    int a[n];
	    for(int i=0;i<n;i++){
	        cin>>a[i];
	    }
	    sort(a,a+n);
	    long long cl=0,cs=0;
	    for(int i=0;i<n;i++){
	        if(a[i]==a[0]) cs++;
	        if(a[i]==a[n-1]) cl++;
	    }
	    if(cs!=n) cout<<2*cs*cl<<endl;
	    else cout<<n*(n-1)<<endl;
	}

}

File: Problems/Graph/Day-11/sol/ishanrajsingh/Solution1.py
# Submission: https://codeforces.com/contest/580/submission/356792507
# Approach: build tree and DFS tracking consecutive cats, count leaves reachable with ≤m cats
# Runtime: O(n)
import sys
sys.setrecursionlimit(3000000)
input=sys.stdin.readline

n,m=map(int,input().split())
cats=list(map(int,input().split()))
g=[[] for _ in range(n)]
for _ in range(n-1):
    u,v=map(int,input().split())
    u-=1; v-=1
    g[u].append(v)
    g[v].append(u)

res=0
visited=[False]*n
def dfs(u,consec):
    global res
    visited[u]=True
    if cats[u]==1: consec+=1
    else: consec=0
    if consec>m: return
    leaf=True
    for v in g[u]:
        if not visited[v]:
            leaf=False
            dfs(v,consec)
    if leaf: res+=1

dfs(0,0)
print(res)
